From 74366e8cdf52cf6e73488928e4e4b05204c3d285 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Fri, 13 Nov 2020 16:53:18 +0100
Subject: [PATCH 1/2] Add pkgconfig file

Without it, -lapplauncherd is undefined and it can not be used to
compile other applications with it. I needed this to get
mapplauncherd-qt to compile
---
 .gitignore                     |  1 +
 CMakeLists.txt                 | 12 +++++++++---
 src/launcherlib/CMakeLists.txt | 25 +++++++++++++++++--------
 3 files changed, 27 insertions(+), 11 deletions(-)

diff --git a/.gitignore b/.gitignore
index ced4d8c..2a0695c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -21,3 +21,4 @@ tests/perftests/hello_no_launcher/hello_no_launcher
 tests/perftests/launcher_perf.txt
 build-stamp
 configure-stamp
+build
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05bb06b..5ea86c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,16 @@
-project(Applauncherd)
+cmake_minimum_required(VERSION 3.0)
+cmake_policy(VERSION 3.0)
 
-cmake_minimum_required(VERSION 2.6)
-cmake_policy(VERSION 2.6)
+project(Applauncherd
+	VERSION 4.1.43
+	DESCRIPTION "A daemon that helps to launch applications faster")
+
+find_package(ECM REQUIRED NO_MODULE)
+set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 
 include(FeatureSummary)
 include(GNUInstallDirs)
+include(ECMGeneratePkgConfigFile)
 
 option(INSTALL_SYSTEMD_UNITS "Install systemd unit files" ON)
 
diff --git a/src/launcherlib/CMakeLists.txt b/src/launcherlib/CMakeLists.txt
index 5396614..fd3b945 100644
--- a/src/launcherlib/CMakeLists.txt
+++ b/src/launcherlib/CMakeLists.txt
@@ -1,5 +1,3 @@
-include(GNUInstallDirs)
-
 set(COMMON ${CMAKE_HOME_DIRECTORY}/src/common)
 
 # Find systemd
@@ -24,13 +22,24 @@ set(HEADERS appdata.h booster.h connection.h daemon.h logger.h launcherlib.h
 link_libraries(${LIBDL} "-L/lib -lcap")
 
 # Set executable
-add_library(applauncherd MODULE ${SRC} ${MOC_SRC})
-set_target_properties(applauncherd PROPERTIES VERSION 0.1 SOVERSION 0)
-
+add_library(applauncherd SHARED ${SRC} ${MOC_SRC})
 target_link_libraries(applauncherd ${SYSTEMD_LIBRARIES})
 
+set_target_properties(applauncherd PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR})
+
+# Configure and install the pkgconfig file
+ecm_generate_pkgconfig_file(
+	BASE_NAME applauncherd
+	FILENAME_VAR applauncherd
+	DESCRIPTION ${PROJECT_DESCRIPTION}
+	INSTALL)
 
 # Add install rule
-install(TARGETS applauncherd DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
-install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/applauncherd
-  PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+install(TARGETS applauncherd
+    LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
+install(FILES ${HEADERS}
+    DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/applauncherd
+    COMPONENT Devel
+    PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
-- 
2.23.0


From 3b6123991e394a31ad2bc81a8d26731933737662 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sat, 17 Apr 2021 19:10:25 +0200
Subject: [PATCH 2/2] Packaging: add new pkgconfig file to devel package and
 add new dep on ECM

---
 rpm/mapplauncherd.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rpm/mapplauncherd.spec b/rpm/mapplauncherd.spec
index 477e1b1..8f6f1cf 100644
--- a/rpm/mapplauncherd.spec
+++ b/rpm/mapplauncherd.spec
@@ -11,6 +11,7 @@ Requires(post): /sbin/ldconfig
 Requires(post): /usr/sbin/setcap
 Requires(postun): /sbin/ldconfig
 Requires(pre):  sailfish-setup
+BuildRequires:  extra-cmake-modules
 BuildRequires:  pkgconfig(libshadowutils)
 BuildRequires:  pkgconfig(systemd)
 BuildRequires:  pkgconfig(dbus-1)
@@ -93,6 +94,7 @@ install -D -m 0755 scripts/booster-cgroup-mount %{buildroot}/usr/lib/startup/boo
 %files devel
 %defattr(-,root,root,-)
 %{_includedir}/applauncherd/*
+%{_libdir}/pkgconfig/*.pc
 
 %files cgroup
 %{_unitdir}/booster-cgroup-mount.service
-- 
2.23.0

